### PR TITLE
Expose the authorized keys tmate feature

### PR DESCRIPTION
**Pull request by julienduchesne**
Described here: https://tmate.io/ in "Access Control"
The variable accepts the file content in raw format (with \n) and dumps it into a file which tmate reads

### CODE REVIEW

File: engine/compiler/shell/shell.go [hunk 1]
```diff
@@ -68,6 +68,11 @@ if [ ! -z "${DRONE_TMATE_HOST}" ]; then
 	echo "set -g tmate-server-port $DRONE_TMATE_PORT" >> $HOME/.tmate.conf
 	echo "set -g tmate-server-rsa-fingerprint $DRONE_TMATE_FINGERPRINT_RSA" >> $HOME/.tmate.conf
 	echo "set -g tmate-server-ed25519-fingerprint $DRONE_TMATE_FINGERPRINT_ED25519" >> $HOME/.tmate.conf
+
+	if [ ! -z "${DRONE_TMATE_AUTHORIZED_KEYS}" ]; then
+		printf "$DRONE_TMATE_AUTHORIZED_KEYS" > $HOME/.tmate.authorized_keys
```
[the remark this answers is not in the document]
Comment (julienduchesne) [reply engine/compiler/shell/shell.go]: Done

**Comment by julienduchesne** [comment]
@bradrydzewski This was tested on our Drone instance and it works well. Users need to have the correct SSH key to connect to the tmate session

**Comment by julienduchesne** [comment]
@bradrydzewski Another thing I noticed is that when you set `DRONE_TMATE_HOST`, it creates the .tmate.conf file even if debug mode is not active. This causes issues in containers that don't have the $HOME environment variable.

So I placed the tmate config creation under the debug mode `if` statement. That way, if there's an issue with the config creation, it only happens in debug mode

**Review by tphoney — APPROVED** [review]

**Comment by tphoney** [comment]
Thanks for the changes @julienduchesne! 

Really appreciated 🎉

**Pull request merged**